tests/unit/s2n_openssl_x509_test.c

```diff
@@ -76,5 +79,130 @@ int main(int argc, char** argv)
         }
     }
 
+    /* s2n_openssl_x509_get_cert_info */
+    /* clang-format off */
+    struct {
+        const char *key_type;
+        const char *signature;
+        const char *key_size;
+        const char *digest;
+        int expected_signature_nid;
+        int expected_digest_nid;
+    } test_cases[] = {
+        { 
+            .key_type = "ec",
+            .signature = "ecdsa",
+            .key_size = "p384",
+            .digest = "sha256",
```

Comment:
Nit: These should really probably be constants / enums. You're going to find your typos at runtime, not in your IDE or at compile time.